zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/ZiplineLoader.kt

```diff
@@ -52,48 +125,145 @@ class ZiplineLoader(
       concurrentDownloadsSemaphore = Semaphore(value)
     }
 
-  // TODO add schema version checker and automigration for non-Android platforms
-  private val cache = createZiplineCache(
-    driver = cacheDbDriver,
-    fileSystem = cacheFileSystem,
-    directory = cacheDir,
-    maxSizeInBytes = cacheMaxSizeInBytes.toLong(),
-    nowMs = nowMs
+  fun withNetwork(): ZiplineLoader = ZiplineLoader(
+    dispatcher = dispatcher,
+    httpClient = httpClient,
+    fetchers = fetchers + listOf(
+      HttpFetcher(httpClient = httpClient)
+    )
+  )
+
+  fun withCachedNetwork(
+    cacheDbDriver: SqlDriver, // SqlDriver is already initialized to the platform and SQLite DB on disk
+    cacheDir: Path,
+    cacheFileSystem: FileSystem,
+    cacheMaxSizeInBytes: Int = 100 * 1024 * 1024,
+    nowMs: () -> Long, // 100 MiB
+  ): ZiplineLoader = ZiplineLoader(
+    dispatcher = dispatcher,
+    httpClient = httpClient,
+    fetchers = fetchers + listOf(
```

Comment:
This is where I don’t like the current API. There’s an implied precedence of the calls now.

Reply:
All these with-ers need to be removed. There's no reason this type should have any knowledge of fetcher subtypes. Fetcher precedence is entirely something that should be explicit